internal/docker/builder.go

```diff
@@ -354,7 +354,10 @@ func (d *Builder) construct(bprint b.Blueprint) (errs []error) {
 // construct this homeserver and execute its instructions, keeping the container alive.
 func (d *Builder) constructHomeserver(blueprintName string, runner *instruction.Runner, hs b.Homeserver, networkID string) result {
 	contextStr := fmt.Sprintf("%s.%s.%s", d.Config.PackageNamespace, blueprintName, hs.Name)
-	d.log("%s : constructing homeserver...\n", contextStr)
+	baseImageURI := d.Config.BaseImageURIs[hs.Name]
+	hs.BaseImageURI = &baseImageURI
+	d.log("%s : constructing homeserver %s using baseimage %s...\n", contextStr, hs.Name, baseImageURI)
```

Comment:
I feel like we should do either not log the base image, or move the check on whether to use the default image here (which at a quick glance looks like it should be OK). Otherwise it will lead to fairly confusing logging if `COMPLEMENT_BASE_IMAGE_HSX` isn't provided.

Reply:
Hm, somehow forgot to push, because I've already removed that. :)